agent.go

```diff
@@ -229,18 +234,22 @@ func (agent *agentS) head(url string) (string, error) {
 }
 
 func (agent *agentS) request(url string, method string, data interface{}) (string, error) {
-	return agent.fullRequestResponse(url, method, data, nil, "")
+	ctx, cancel := context.WithTimeout(context.Background(), agent.clientTimeout)
+	defer cancel()
+	return agent.fullRequestResponse(ctx, url, method, data, nil, "")
 }
 
-func (agent *agentS) requestResponse(url string, method string, data interface{}, ret interface{}) (string, error) {
-	return agent.fullRequestResponse(url, method, data, ret, "")
+func (agent *agentS) announceRequest(url string, method string, data interface{}, ret *agentResponse) (string, error) {
+	return agent.fullRequestResponse(context.Background(), url, method, data, ret, "")
 }
 
 func (agent *agentS) requestHeader(url string, method string, header string) (string, error) {
-	return agent.fullRequestResponse(url, method, nil, nil, header)
+	ctx, cancel := context.WithTimeout(context.Background(), agent.clientTimeout)
```

Comment:
Same as above, shall we elaborate on using different timeout?

Reply:
Agree